File: mnemosyne/mnemosyne/embedded_in_C/main_wdgt.c
//
// main_wdgt.c <Peter.Bienstman@UGent.be>
//
//   Template, to be replaced by functions which actually do something useful
//   in a specific client.
//

#include <stdio.h>
#include <string.h>

void main_wdgt_set_window_title(char* text)
{
  printf("set_window_title: %s\n", text);
}


void main_wdgt_show_information(char* text)
{
  printf("show_information: %s\n", text);
}


int main_wdgt_show_question(char* text, char* option_0, char* option_1,
                            char* option_2)
{
  printf("show_question: %s, %s, %s, %s\n", text, option_0, option_1,
         option_2);
  // As an example, we hardcode the reply.
  return 0;
}


void main_wdgt_show_error(char* text)
{
  printf("show_error: %s\n", text);
}


void main_wdgt_get_filename_to_open(char* path, char* filter, char *caption,
                                    char* filename, int str_size)
{
  printf("get_filename_to_open: %s, %s, %s\n", path, filter, caption);
  // As an example, we hardcode the reply.
  strncpy(filename, "example_path", str_size);
}


void main_wdgt_get_filename_to_save(char* path, char* filter, char *caption,
                                    char* filename, int str_size)
// Should warn about overwriting existing file.
{
  printf("get_filename_to_save: %s, %s, %s\n", path, filter, caption);
  // As an example, we hardcode the reply.
  strncpy(filename, "example_path", str_size);
}


void main_wdgt_set_status_bar_message(char* text)
{
  printf("set_status_bar_message: %s\n", text);
}


void main_wdgt_set_progress_text(char* text)
{
  printf("set_progress_text: %s\n", text);
}


void main_wdgt_set_progress_range(int max)
{
  printf("set_progress_range: %d\n", max);
}


void main_wdgt_set_progress_update_interval(int interval)
{
  printf("set_progress_update_interval: %d\n", interval);
}


void main_wdgt_increase_progress(int value)
{
  printf("increase_progress: %d\n", value);
}

void main_wdgt_set_progress_value(int value)
{
  printf("set_progress_value: %d\n", value);
}


void main_wdgt_close_progress()
{
  printf("close_progress\n");
}


void main_wdgt_enable_edit_current_card(int is_enabled)
{
  printf("enable_edit_current_card: %d\n", is_enabled);
}


void main_wdgt_enable_delete_current_card(int is_enabled)
{
  printf("enable_delete_current_card: %d\n", is_enabled);
}


void main_wdgt_enable_browse_cards(int is_enabled)
{
  printf("enable_browse_cards: %d\n", is_enabled);
}

File: mnemosyne/mnemosyne/embedded_in_C/main_wdgt.py
#
# main_wdgt.py <Peter.Bienstman@UGent.be>
#

import _main_wdgt as _
from mnemosyne.libmnemosyne.ui_components.main_widget import MainWidget


class MainWdgt(MainWidget):

    def set_window_title(self, text):
        _.set_window_title(text.encode("utf-8"))
        
    def show_information(self, text):
        _.show_information(text.encode("utf-8"))
        
    def show_question(self, text, option0, option1, option2=""):
        return _.show_question(text.encode("utf-8"),
            option0.encode("utf-8"), option1.encode("utf-8"),
            option2.encode("utf-8"))
    
    def show_error(self, text):
        _.show_information(text.encode("utf-8"))

    def get_filename_to_open(self, path, filter, caption=""):
        return _.get_filename_to_open(path.encode("utf-8"),
            filter.encode("utf-8"), caption.encode("utf-8"))
    
    def get_filename_to_save(self, path, filter, caption=""):
        return _.get_filename_to_save(path.encode("utf-8"),
            filter.encode("utf-8"), caption.encode("utf-8"))
   
    def set_status_bar_message(self, text):
        _.set_status_bar_message(text.encode("utf-8"))
    
    def set_progress_text(self, text):
        _.set_progress_text(text.encode("utf-8"))
                
    def set_progress_range(self, minimum, maximum):
        _.set_progress_range(minimum, maximum)
        
    def set_progress_update_interval(self, update_interval):
        _.set_progress_update_interval(update_interval)
        
    def set_progress_value(self, value):
        _.set_progress_value(value)
        
    def close_progress(self):
        _.close_progress()

    def enable_edit_current_card(self, is_enabled):
        _.enable_edit_current_card(is_enabled)

    def enable_delete_current_card(self, is_enabled):
        _.enable_delete_current_card(is_enabled)

    def enable_browse_cards(self, is_enabled):      
        _.enable_browse_cards(is_enabled)
